pkg/storage/replica.go

```diff
@@ -5521,6 +5521,11 @@ func evaluateBatch(
 						// commit, we'll need a client-side retry, so we return immediately
 						// to see if we can do a txn coord sender retry instead.
 						returnWriteTooOldErr = true
+					case *roachpb.InitPutRequest:
```

Comment:
I made the exact same change last weekend but didn't end up pushing it because I was struggling to understand a few cases in `TestTxnCoordSenderRetries`. Glad you made this change!